src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorDynamicFileInfoProvider.cs

```diff
@@ -3,60 +3,61 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Composition;
-using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis.Collections;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
 using Microsoft.CodeAnalysis.LanguageServer.LanguageServer;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.CodeAnalysis.Text;
-using Roslyn.LanguageServer.Protocol;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
 
-[Export(typeof(IDynamicFileInfoProvider)), Shared]
+[Shared]
+[Export(typeof(IDynamicFileInfoProvider))]
 [ExportMetadata("Extensions", new string[] { "cshtml", "razor", })]
-internal class RazorDynamicFileInfoProvider : IDynamicFileInfoProvider
+[ExportCSharpVisualBasicStatelessLspService(typeof(RazorDynamicFileInfoProvider))]
```

Comment:
David can confirm, but I think having this be exported as a LSP service and a regular one will cause lifetime issues. Or we'll end up with two instances, which might be problematic if you were hoping to have shared state? Either way I think separating into two classes would be clearer.